spec/support/outbox_models.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+Object.const_set('Uuid', Module.new)
+Object.const_set('Id', Module.new)
+Uuid::Outbox = Class.new(ActiveRecord::Base) do
+  def self.name
+    'Uuid::Outbox'
+  end
+
+  def self.table_name
+    'uuid_outboxes'
+  end
+
+  validates_presence_of :identifier, :payload, :aggregate, :aggregate_identifier, :event
+end
+
+Id::Outbox = Class.new(ActiveRecord::Base) do
+  def self.name
+    'Id::Outbox'
+  end
+
+  def self.table_name
+    'id_outboxes'
+  end
+
+  validates_presence_of :identifier, :payload, :aggregate, :aggregate_identifier, :event
+end
+
+Uuid::FakeModel = Class.new(ActiveRecord::Base) do
+  def self.name
+    'Uuid::FakeModel'
+  end
+
+  def self.table_name
+    'uuid_fake_models'
+  end
+
+  validates_presence_of :test_field
+  include ActiveOutbox::Outboxable
+end
+
+Id::FakeModel = Class.new(ActiveRecord::Base) do
+  def self.name
+    'Id::FakeModel'
+  end
+
+  def self.table_name
+    'id_fake_models'
+  end
+
+  validates_presence_of :test_field
+  include ActiveOutbox::Outboxable
+end
+
+def create_migrations
+  id_migrations
+  uuid_migrations
+end
+
+def id_migrations
+  ActiveRecord::Base.connection.create_table :id_fake_models, if_not_exists: true do |t|
```

Comment:
are the tests wrapped in a DB transaction? would be good so that the DB is in a clean state before each run

Reply:
we use database cleaner to achieve a consistent state between runs